build_projects/dotnet-cli-build/BuildFPMToolPreReqs.cs

```diff
@@ -0,0 +1,319 @@
+                    FPMParameters = BuildCmdParameters(configJson, PackageVersion);
+                    Log.LogMessage(MessageImportance.Normal, "Generated RPM paramters:  " + FPMParameters);
+                }
+                else
+                {
+                    throw new IOException($"Could not open the file {ConfigJsonFile} for reading.");
+                }
+            }
+            catch (Exception e)
+            {
+                Log.LogError("Exception while processing RPM paramters: " + e.Message);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        // Update the tokens in the changelog file from the config Json 
+        private void UpdateChangelog(ConfigJson configJson, string package_version)
+        {
+            try
+            {
+                string changelogFile = Path.Combine(InputDir, "templates", "changelog");
+                if (!File.Exists(changelogFile))
+                {
+                    throw new FileNotFoundException($"Expected file {changelogFile} was not found.");
+                }
+                string str = File.ReadAllText(changelogFile);
+                str = str.Replace("{PACKAGE_NAME}", configJson.Package_Name);
+                str = str.Replace("{PACKAGE_VERSION}", package_version);
+                str = str.Replace("{PACKAGE_REVISION}", configJson.Release.Package_Revision);
+                str = str.Replace("{URGENCY}", configJson.Release.Urgency);
+                str = str.Replace("{CHANGELOG_MESSAGE}", configJson.Release.Changelog_Message);
+                str = str.Replace("{MAINTAINER_NAME}", configJson.Maintainer_Name);
+                str = str.Replace("{MAINTAINER_EMAIL}", configJson.Maintainer_Email);
+                // The date format needs to be like Wed May 17 2017
+                str = str.Replace("{DATE}", DateTime.UtcNow.ToString("ddd MMM dd yyyy")); 
+                File.WriteAllText(changelogFile, str);
+            }
+            catch (Exception e)
+            {
+                Log.LogError("Exception while updating the changelog file: " + e.Message);
+            }
+        }
+
+        public void UpdateCopyRight(ConfigJson configJson)
+        {
+            try
+            {
+                // Update the tokens in the copyright file from the config Json 
+                string copyrightFile = Path.Combine(InputDir, "templates", "copyright");
+                if (!File.Exists(copyrightFile))
+                {
+                    throw new FileNotFoundException($"Expected file {copyrightFile} was not found.");
+                }
+                string str = File.ReadAllText(copyrightFile);
+                str = str.Replace("{COPYRIGHT_TEXT}", configJson.CopyRight);
+                str = str.Replace("{LICENSE_NAME}", configJson.License.Type);
+                str = str.Replace("{LICENSE_NAME}", configJson.License.Type);
+                str = str.Replace("{LICENSE_TEXT}", configJson.License.Full_Text);
```

Comment:
Need more time to look into it, since it involves legal, also, that is what runtime package does. I don't think it is a blocking issue today